src/EPR.Payment.Mopup.UnitTests/Services/PaymentsServiceTests.cs

```diff
@@ -0,0 +1,181 @@
+            var mapperConfig = new MapperConfiguration(cfg =>
+            {
+                cfg.AddProfile<PaymentProfile>();
+            });
+            _mapper = mapperConfig.CreateMapper();
+
+            _paymentRepositoryMock = new Mock<IPaymentsRepository>();
+            _httpGovPayServiceMock = new Mock<IHttpGovPayService>();
+            _loggerMock = new Mock<ILogger<PaymentsService>>();
+
+            _paymentsService = new PaymentsService(
+                _paymentRepositoryMock.Object,
+                _httpGovPayServiceMock.Object,
+                _loggerMock.Object,
+                _mapper
+                );
+
+        }
+
+        [TestMethod, AutoMoqData]
+        public async Task UpdatePaymentsAsync_ShouldUpdatePaymentStatus(
+            [Frozen] List<Common.Data.DataModels.Payment> _payments,
+            [Frozen] PaymentStatusResponseDto _paymentStatusResponseDto)
+        {
+            // Arrange
+            _paymentRepositoryMock
+                .Setup(x => x.GetPaymentsByStatusAsync(Status.InProgress, It.IsAny<CancellationToken>()))
+                .ReturnsAsync(_payments);
+
+            List<PaymentDto> _paymentDtos = _mapper.Map<List<PaymentDto>>(_payments);
+
+            foreach (var paymentDto in _paymentDtos)
+            {
+                _paymentStatusResponseDto!.State!.Status = "success";
+                _paymentStatusResponseDto!.State!.Code = "";
+                _paymentStatusResponseDto!.State!.Message = "Payment Successful";
+
+                _httpGovPayServiceMock
+                    .Setup(x => x.GetPaymentStatusAsync(paymentDto.GovpayPaymentId!, It.IsAny<CancellationToken>()))
+                    .ReturnsAsync(_paymentStatusResponseDto);
+
+                paymentDto.Status = PaymentStatusMapper.GetPaymentStatus(_paymentStatusResponseDto!.State!.Status!, _paymentStatusResponseDto.State.Code);
+
+                UpdatePaymentRequestDto _updatePaymentRequestDto = _mapper.Map<UpdatePaymentRequestDto>(paymentDto);
+
+                var entity = _payments.SingleOrDefault(x => x.Id == paymentDto.Id);
+
+                _mapper.Map(_updatePaymentRequestDto, entity);
+            }
+
+            // Act
+            await _paymentsService.UpdatePaymentsAsync();
+
+            // Assert
+            using (new AssertionScope())
+            {
+                _paymentRepositoryMock.Verify(x => x.UpdatePaymentStatusAsync(It.IsAny<Common.Data.DataModels.Payment>(), It.IsAny<CancellationToken>()), Times.AtLeastOnce);
+            }
+        }
+
+        [TestMethod, AutoMoqData]
+        public async Task UpdatePaymentsAsync_ShouldThrowException_WhenGovPayPaymentIdIsNull(
+            [Frozen] List<Common.Data.DataModels.Payment> _payments)
+        {
+            // Arrange
+            _payments.ForEach(s => s.GovpayPaymentId = null);
+
+            _paymentRepositoryMock
+                .Setup(repo => repo.GetPaymentsByStatusAsync(Common.Enums.Status.InProgress, It.IsAny<CancellationToken>()))
```

Comment:
Common.Enums is redundant. can be removed.

Advice - Instead of injecting a frozen instance of a list, we can create a list with single element having GovPaypaymentId as null. Need not necessarily change